src/Microsoft.Data.SqlClient/netcore/src/Microsoft/Data/SqlClient/TdsParser.cs

```diff
@@ -13088,17 +13088,17 @@ bool writeDataSizeToSnapshot
                 ||
                 (buff.Length >= offst + len)
                 ||
-                (buff.Length == (startOffsetByteCount >> 1) + 1),
+                (buff.Length >= (startOffsetByteCount >> 1) + 1),
                 "Invalid length sent to ReadPlpUnicodeChars()!"
             );
             charsLeft = len;
 
             // If total length is known up front, the length isn't specified as unknown 
             // and the caller doesn't pass int.max/2 indicating that it doesn't know the length
```

Comment:
I think with this change, the comment will need to be changed as well - for this check we no longer care what the caller requested.

Reply:
Haha, I'd like something that explains what the check is doing. Right now, it says that the if statment is checking for int.max/2 from the caller (ie, `len`), but the if statement no longer checks `len`. Even if it's not the perfect explanation what's going on, it'd still be good to get some understanding of what the check tries to do or why it's being made. Basically, I just don't want stale/inaccurate information in the comments - it'll make our lives a lot easier if the code and comments agree with each other.

Alternatively, if you think the comment is still accurate as-is, let me know and maybe we can come up with a better wording to explain the behavior.

Reply:
"Invalid length sent to ReadPlpUnicodeChars()!" to "Too short buffer passed to ReadPlpUnicodeChars()!" ?

or for the other one:
```csharp
// If total data length is known up front from the plp header by being not SQL_PLP_UNKNOWNLEN
//  and the number of chars required is less than int.max/2 allocate the entire buffer now to avoid
//  later needing to repeatedly allocate new target buffers and copy data as we discover new data
```

Reply:
The other one - the line I added the comment to. I think what you suggested is a good revision 👍